src/thor/trace_route_action.cc

```diff
@@ -91,9 +91,15 @@ void thor_worker_t::trace_route(Api& request) {
     // If non-exact shape points are used, then we need to correct this shape by sending them
     // through the map-matching algorithm to snap the points to the correct shape
     case ShapeMatch::map_snap:
```

Comment:
you'll want to do this for the other cases in the switch statement as well.

Reply:
@kevinkreiser you're referring to the other call to `map_match` in `ShapeMatch::walk_or_snap`, right? I think the other cases `ShapeMatch::edge_walk` or `ShapeMatch::walk_or_snap => route_match` either don't do candidate search or correctly handle the exceptions. I'll push a change in just a sec and you can let me know if I'm missing anything else.

Reply:
correct you got it there are two calls to the "snap" implementation, those both need this, which i see you got, thanks!